From 1bf7563faeaa7de701d8b2e1d1368f89505c4762 Mon Sep 17 00:00:00 2001
From: Michael Weghorn <m.weghorn@posteo.de>
Date: Thu, 21 May 2020 14:19:32 +0200
Subject: [PATCH] skel.bashrc: Increase HISTSIZE to 2000, don't set
 HISTFILESIZE
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Setting HISTFILESIZE not only sets the value for the environment
variable, but immediately truncates the bash history file,
'~/.bash_history'.

Setting a value in the skel file therefore effectively
makes it impossible for users to simply override that value
by setting it to another value in '~/.bash_aliases', which is
usually regarded as the recommended way to set user-specific
values (since that file is sourced in '~/.bashrc' as provided
by the skeleton file). Thus, the users would have to edit
'~/.bashrc' in addition (and know that this is needed, which
is probably rather unexpected for most users).

To avoid this and allow the user to simply set the desired
history length via environment variable in '~/.bash_aliases',
only set the 'HISTSIZE' variable, which is what 'HISTFILESIZE'
defaults to when not set explicitly.

From bash's man page:

>    HISTFILESIZE
>           The maximum number of  lines  contained  in  the
>           history  file.  When this variable is assigned a
>           value, the history file is truncated, if  neces‐
>           sary,  to  contain  no  more than that number of
>           lines by removing the oldest entries.  The  his‐
>           tory  file  is also truncated to this size after
>           writing it when a shell exits.  If the value  is
>           0,  the  history file is truncated to zero size.
>           Non-numeric values and numeric values less  than
>           zero inhibit truncation.  The shell sets the de‐
>           fault value to the value of HISTSIZE after read‐
>           ing any startup files.

By setting 'HISTSIZE' to 2000 and not setting HISTFILESIZE
in the skeleton file, the default size of the history file
remains 2000 lines, but it can easily be overriden by users.

This fixes Debian bug #741950.
---
 skel.bashrc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/skel.bashrc b/skel.bashrc
index 9360f69..2e7e3b3 100644
--- a/skel.bashrc
+++ b/skel.bashrc
@@ -16,8 +16,8 @@ HISTCONTROL=ignoreboth
 shopt -s histappend
 
 # for setting history length see HISTSIZE and HISTFILESIZE in bash(1)
-HISTSIZE=1000
-HISTFILESIZE=2000
+# (HISTFILESIZE defaults to HISTSIZE)
+HISTSIZE=2000
 
 # check the window size after each command and, if necessary,
 # update the values of LINES and COLUMNS.
-- 
2.26.2

